linkerd/core/src/main/scala/io/buoyant/linkerd/Svc.scala

```diff
@@ -58,9 +62,42 @@ trait StaticSvc { self: Svc =>
   @JsonIgnore
   def pathParams = PerPathParams(configs.map { config =>
     PathParams(config.prefix, config.params)
-  })
+  }, None)
 }
 
 class StaticSvcImpl(val configs: Seq[SvcPrefixConfig]) extends Svc with StaticSvc
 
-class SvcPrefixConfig(val prefix: PathMatcher) extends SvcConfig
\ No newline at end of file
+class SvcPrefixConfig(val prefix: PathMatcher) extends SvcConfig
+
+trait FileSvc { self: Svc =>
+  val serviceFile: File
+
+  @JsonIgnore
+  private[this] val path = serviceFile.path
+
+  @JsonIgnore
+  private[this] lazy val watcher = Watcher(path.getParent)
+
+  @JsonIgnore
+  private[this] def configsAct: Activity[Seq[SvcPrefixConfig]] = {
+    watcher.children.flatMap { children =>
+      children.get(path.getFileName.toString) match {
+        case Some(file: Watcher.File.Reg) => file.data
+        case _ => Activity.pending
```

Comment:
This can put the Activity back into a pending state if the file is deleted.  Instead, I think we probably want to either go into a failed state or use the default SvcPrefixConfig or something.